test/jasmine/tests/parcoords_test.js

```diff
@@ -0,0 +1,900 @@
+
+            function restyleDimension(key, setterValue) {
+
+                // array values need to be wrapped in an array; unwrapping here for value comparison
+                var value = Lib.isArray(setterValue) ? setterValue[0] : setterValue;
+
+                return function() {
+                    return Plotly.restyle(gd, 'dimensions[2].' + key, setterValue).then(function() {
+                        expect(gd.data[0].dimensions[2][key]).toEqual(value, 'for dimension attribute \'' + key + '\'');
+                    });
+                };
+            }
+
+            restyleDimension('label', 'new label')()
+                .then(restyleDimension('tickvals', [[0, 0.1, 0.4, 1, 2]]))
+                .then(restyleDimension('ticktext', [['alpha', 'gamma', 'beta', 'omega', 'epsilon']]))
+                .then(restyleDimension('tickformat', '4s'))
+                .then(restyleDimension('range', [[0, 2]]))
+                .then(restyleDimension('constraintrange', [[0, 1]]))
+                .then(restyleDimension('values', [[0, 0.1, 0.4, 1, 2, 0, 0.1, 0.4, 1, 2]]))
+                .then(restyleDimension('visible', false))
+                .then(done);
+        });
+
+        it('Calling `Plotly.restyle` with an object should amend the preexisting parcoords', function(done) {
+
+            var newStyle = Lib.extendDeep({}, mockCopy.data[0].line);
+            newStyle.colorscale = 'Viridis';
+            newStyle.reversescale = false;
+
+            Plotly.restyle(gd, {line: newStyle}).then(function() {
+
+                expect(gd.data.length).toEqual(1);
+
+                expect(gd.data[0].line.colorscale).toEqual('Viridis');
+                expect(gd.data[0].line.reversescale).toEqual(false);
+                expect(gd.data[0].dimensions.length).toEqual(11);
+                expect(gd.data[0].line.cmin).toEqual(-4000);
+                expect(gd.data[0].dimensions[0].constraintrange).toBeDefined();
+                expect(gd.data[0].dimensions[0].constraintrange).toEqual([100000, 150000]);
+                expect(gd.data[0].dimensions[1].constraintrange).not.toBeDefined();
+
+                done();
+            });
+
+
+        });
+
+        it('Should emit a \'plotly_restyle\' event', function(done) {
+
+            var tester = (function() {
+
+                var eventCalled = false;
+
+                return {
+                    set: function(d) {eventCalled = d;},
+                    get: function() {return eventCalled;}
+                };
+            })();
+
+            gd.on('plotly_restyle', function() {
+                tester.set(true);
+            });
+
+            expect(tester.get()).toBe(false);
+            Plotly.restyle(gd, 'line.colorscale', 'Viridis')
+                .then(window.setTimeout(function() {
+                    expect(tester.get()).toBe(true);
+                    done();
+                }, 0));
+
+        });
+
+        it('Should emit a \'plotly_hover\' event', function(done) {
```

Comment:
Maybe somebody else could pull down this branch and run `npm run test-jasmine -- tests/parcoords_test.js` to check?